docs: Tests fail fresh clone of master branch

### What was unclear or otherwise insufficient?

I am missing some step to setup my machine for contributing to this library.

Tests are failing due to missing dependencies related to Svelte and WebUI tests.

I followed the steps in [development.md](https://node-llama-cpp.withcat.ai/guide/development) including the `cmake` dependencies.

The [linked page for cmake](https://github.com/cmake-js/cmake-js#installation) is not clear so what I did was:
```
npm install -g cmake-js
```
And:
```
cmake-js install
```

I also had cmake installed separately via Homebrew.

I am using the Node and npm versions @giladgd is using:

Node `24.5.0` and npm `11.5.1`

Have I missed a step?

Here are the failed test outputs:

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Suites 5 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯

 FAIL  llama/llama.cpp/tools/server/webui/tests/client/page.svelte.test.ts [ llama/llama.cpp/tools/server/webui/tests/client/page.svelte.test.ts ]
TSConfckParseError: failed to resolve "extends":"./.svelte-kit/tsconfig.json" in /Users/albert/Developer/node-llama-cpp/llama/llama.cpp/tools/server/webui/tsconfig.json
  Plugin: vite:esbuild
  File: /Users/albert/Developer/node-llama-cpp/llama/llama.cpp/tools/server/webui/tests/client/page.svelte.test.ts
 ❯ resolveExtends node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:18845:8
 ❯ parseExtends node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:18780:24
 ❯ parse$e node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:18638:23
 ❯ loadTsconfigJsonForFile node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:19433:20
 ❯ transformWithEsbuild node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:19194:30
 ❯ TransformPluginContext.transform node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:19297:24
 ❯ PluginContainer.transform node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:49098:18
 ❯ loadAndTransform node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:51931:27

Caused by: Error: Cannot find module './.svelte-kit/tsconfig.json'
Require stack:
- /Users/albert/Developer/node-llama-cpp/llama/llama.cpp/tools/server/webui/tsconfig.json
 ❯ resolveExtends node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:18833:14
 ❯ parseExtends node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:18780:24
 ❯ parse$e node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:18638:23
 ❯ loadTsconfigJsonForFile node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:19433:20
 ❯ transformWithEsbuild node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:19194:30
 ❯ TransformPluginContext.transform node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:19297:24
 ❯ PluginContainer.transform node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:49098:18
 ❯ loadAndTransform node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:51931:27

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/5]⎯

 FAIL  llama/llama.cpp/tools/server/webui/tests/e2e/demo.test.ts [ llama/llama.cpp/tools/server/webui/tests/e2e/demo.test.ts ]
 FAIL  llama/llama.cpp/tools/server/webui/tests/server/demo.spec.ts [ llama/llama.cpp/tools/server/webui/tests/server/demo.spec.ts ]
 FAIL  llama/llama.cpp/tools/server/webui/src/lib/utils/latex-protection.test.ts [ llama/llama.cpp/tools/server/webui/src/lib/utils/latex-protection.test.ts ]
 FAIL  llama/llama.cpp/tools/server/webui/src/lib/utils/model-names.test.ts [ llama/llama.cpp/tools/server/webui/src/lib/utils/model-names.test.ts ]
TSConfckParseError: failed to resolve "extends":"./.svelte-kit/tsconfig.json" in /Users/albert/Developer/node-llama-cpp/llama/llama.cpp/tools/server/webui/tsconfig.json
  Plugin: vite:esbuild
  File: /Users/albert/Developer/node-llama-cpp/llama/llama.cpp/tools/server/webui/tests/e2e/demo.test.ts
 ❯ resolveExtends node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:18845:8
 ❯ parseExtends node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:18780:24
 ❯ getParsedDeep node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:18662:4
 ❯ parse$e node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:18635:13
 ❯ loadTsconfigJsonForFile node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:19433:20
 ❯ transformWithEsbuild node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:19194:30
 ❯ TransformPluginContext.transform node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:19297:24
 ❯ PluginContainer.transform node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:49098:18
 ❯ loadAndTransform node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:51931:27

Caused by: Error: Cannot find module './.svelte-kit/tsconfig.json'
Require stack:
- /Users/albert/Developer/node-llama-cpp/llama/llama.cpp/tools/server/webui/tsconfig.json
 ❯ resolveExtends node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:18833:14
 ❯ parseExtends node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:18780:24
 ❯ getParsedDeep node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:18662:4
 ❯ parse$e node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:18635:13
 ❯ loadTsconfigJsonForFile node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:19433:20
 ❯ transformWithEsbuild node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:19194:30
 ❯ TransformPluginContext.transform node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:19297:24
 ❯ PluginContainer.transform node_modules/vite/dist/node/chunks/dep-CHZK6zbr.js:49098:18

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/5]⎯


 Test Files  5 failed | 60 passed | 1 skipped (66)
      Tests  285 passed | 6 skipped (291)
   Start at  20:06:32
   Duration  527.96s (transform 615ms, setup 33ms, collect 2.99s, tests 524.09s, environment 0ms, prepare 559ms)
```

### Recommended Fix

Update the docs with any missing steps for development.

(Or if I have misread/misunderstood the docs I can propose some clarification so that others like me don't make the same mistake.)

### Additional Context

_No response_

### Are you willing to resolve this issue by submitting a Pull Request?

Yes, I have the time, but I don't know how to start. I would need guidance.

--- a/docs/guide/development.md
+++ b/docs/guide/development.md
@@ -8,7 +8,7 @@ This document describes how to set up your development environment to contribute
 ## Prerequisites
 - [Git](https://git-scm.com/). [GitHub's Guide to Installing Git](https://help.github.com/articles/set-up-git) is a good source of information.
 - [Node.js](https://nodejs.org/en/) (v20 or higher)
-- [cmake dependencies](https://github.com/cmake-js/cmake-js#installation:~:text=projectRoot/build%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%5Bstring%5D-,Requirements%3A,-CMake) - make sure the required dependencies of `cmake` are installed on your machine. More info is available [here](https://github.com/cmake-js/cmake-js#installation:~:text=projectRoot/build%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%5Bstring%5D-,Requirements%3A,-CMake) (you don't necessarily have to install `cmake`, just the dependencies)
+- [cmake dependencies](https://github.com/cmake-js/cmake-js#:~:text=projectRoot/build%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%5Bstring%5D-,Requirements%3A,-CMake) - make sure the required dependencies of `cmake` are installed on your machine. More info is available [here](https://github.com/cmake-js/cmake-js#installation:~:text=projectRoot/build%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%5Bstring%5D-,Requirements%3A,-CMake) (you don't necessarily have to install `cmake`, just the dependencies)
 
 ## Setup
 1. [Fork `node-llama-cpp` repo](https://github.com/withcatai/node-llama-cpp/fork)

--- a/vitest.config.ts
+++ b/vitest.config.ts
@@ -1,7 +1,11 @@
-import {defineConfig} from "vitest/config";
+import {defineConfig, configDefaults} from "vitest/config";
 
 export default defineConfig({
     test: {
+        exclude: [
+            ...configDefaults.exclude,
+            "./llama"
+        ],
         pool: "forks",
         maxWorkers: 1,
         minWorkers: 1,

